### ISSUE TITLE
Low text contrast with pkgdown dark theme

### ISSUE THREAD

**Issue by Bisaloo** [rstudio/DT#1144]
![image](https://github.com/user-attachments/assets/9a721799-7bf0-40ea-bf98-62e9276aa186)

https://epiverse-trace.github.io/epiparameter/articles/database.html (different theme but same issue)

The latest pkgdown version enables dark themes (by setting `light-switch: true` in `_pkgdown.yml`. But the dark background causes contrast issues with the text in the DT because text color is hardcoded in DT:

https://github.com/rstudio/DT/blob/4e63a4664d9caa9fb9d4fb17061ce5eab94bec18/inst/htmlwidgets/lib/datatables/css/jquery.dataTables.extra.css#L26-L28

Are these lines still necessary? Is it possible to inherit from the body text colour, as mentioned in https://github.com/DataTables/DataTablesSrc/issues/157#issuecomment-600526294?

---

By filing an issue to this repo, I promise that

- [ ] I have fully read the issue guide at https://yihui.org/issue/.
- [ ] I have provided the necessary information about my issue.
    - If I'm asking a question, I have already asked it on Stack Overflow or RStudio Community, waited for at least 24 hours, and included a link to my question there.
    - If I'm filing a bug report, I have included a minimal, self-contained, and reproducible example, and have also included `xfun::session_info('DT')`. I have upgraded all my packages to their latest versions (e.g., R, RStudio, and R packages), and also tried the development version: `remotes::install_github('rstudio/DT')`.
    - If I have posted the same issue elsewhere, I have also mentioned it in this issue.
- [ ] I have learned the Github Markdown syntax, and formatted my issue correctly.

I understand that my issue may be closed if I don't fulfill my promises.

**Comment by yihui** [comment]
My gut feeling is that we should use `color: inherit` instead of hardcoding it to `#333`, but someone needs to verify if that would fix https://github.com/rstudio/DT/issues/447.

**Comment by glin** [comment]
Agreed with double checking #447, because I think that would still be an issue in the latest RStudio IDE. https://github.com/rstudio/rstudio/issues/6489 never got a resolution as far as I know

Having hit this same issue in reactable (https://github.com/glin/reactable/issues/21), what I did was inherit text colors by default. And then for RStudio R Notebooks specifically, hack in a terrible workaround to explicitly set the widget's text color ONLY when running inside an R Notebook:
```r
reactable_html <- function(id, style, class, ...) {
  # Set text color in R Notebooks to prevent contrast issues when
  # using a dark editor theme and htmltools 0.4.0.
  if (isTRUE(getOption("rstudio.notebook.executing"))) {
    style <- paste0("color: #333;", style)
  }
  ...
}
```
